pkg/pagerduty/pagerduty.go

```diff
@@ -0,0 +1,100 @@
+package pagerduty
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"time"
+
+	pkgerrors "github.com/pkg/errors"
+
+	sdk "github.com/PagerDuty/go-pagerduty"
+)
+
+const (
+	InvalidInputParamsErrorCode = 2001
+	pagerDutyTimeout            = time.Second * 30
+)
+
+// PagerDuty will hold all of the required fields for any PagerDuty Operation
+type PagerDuty struct {
+	// c is the PagerDuty client
+	c *sdk.Client
+	// userEmail is the user that will run the commands
+	userEmail string
+}
+
+// NewWithToken is similar to New but you only need to supply to authentication token to start
+// The token can be created using the docs https://support.pagerduty.com/docs/api-access-keys#section-generate-a-user-token-rest-api-key
+func NewWithToken(authToken string) (PagerDuty, error) {
+	c := sdk.NewClient(authToken)
+	return New(c)
+}
+
+// New will create a PagerDuty struct with all of the required fields
+func New(client *sdk.Client) (PagerDuty, error) {
+	opts := sdk.GetCurrentUserOptions{
+		Includes: []string{},
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), pagerDutyTimeout)
+	defer cancel()
+	user, err := client.GetCurrentUserWithContext(ctx, opts)
+	if err != nil {
+		perr := sdk.APIError{}
+		if errors.As(err, &perr) {
+			if perr.StatusCode == http.StatusUnauthorized {
+				return PagerDuty{}, pkgerrors.Wrap(err, "the authToken that was provided is invalid")
+			}
+		}
+
+		return PagerDuty{}, pkgerrors.Wrap(err, "could not retrieve the current user")
+	}
+
+	resp := PagerDuty{
+		c:         client,
+		userEmail: user.Email,
+	}
+
+	return resp, nil
+}
+
+// MoveToEscalationPolicy will move the alerts EscalationPolicy to the new EscalationPolicy
+func (p PagerDuty) MoveToEscalationPolicy(incident Incident, escalationPolicy EscalationPolicy) error {
+	if incident.EscalationPolicy == escalationPolicy {
+		return nil
+	}
+	incident.EscalationPolicy = escalationPolicy
+
+	o := []sdk.ManageIncidentsOptions{
+		{
+			ID: incident.ID,
+			EscalationPolicy: &sdk.APIReference{
+				Type: "escalation_policy_reference",
+				ID:   incident.EscalationPolicy.ID,
+			},
+		},
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), pagerDutyTimeout)
+	defer cancel()
+	_, err := p.c.ManageIncidentsWithContext(ctx, p.userEmail, o)
+
+	if err != nil {
+		perr := sdk.APIError{}
+		if errors.As(err, &perr) {
+			switch perr.StatusCode {
+			case http.StatusUnauthorized:
+				return InvalidTokenErr{Err: err}
+			case http.StatusBadRequest:
+				isErrAnInvalidInputErr := perr.APIError.Valid &&
+					perr.APIError.ErrorObject.Code == InvalidInputParamsErrorCode
+				if isErrAnInvalidInputErr {
+					return InvalidInputParamsErr{Err: err}
```

Comment:
A 400 is always a client-side bad request. Do we really need to create the boolean isErrAnInvalidInputErr? On any http.StatusBadRequest we shouldn't retry, so the logic is the same whether the InputParameter are invalid or not

Reply:
I am not retrying, it's just so I can wrap the PD APIError with something I can pass to the user && validate.

extracting to a variable was to try and make the if check smaller

Reply:
sure, I was just wondering why you are untangling this specific error when there is no different logic in place to handle this case. Especially as you return [here]( https://github.com/openshift/configuration-anomaly-detection/pull/4/files/73105185ca69ce1667ab3caca50dd82effad072d#diff-ff48eb69c8e0cf64f068d910259cb2b2aa429edae92bde70980c64d2e290a85fR96) a  generic err object ... But sure at the moment, the result is not used anyways.

Reply:
it makes sense if the CLI wants to do something prettier on the error, and it makes testing easier (I don't need to know in the test the SDK types.

this makes the test be more or less what the library user will take

Reply:
done @Tessg22

Reply:
Where? I don't see the file updated